modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/plugins/fields-sidebar/components/FieldsSidebar.es.js

```diff
@@ -17,20 +17,111 @@ import classNames from 'classnames';
 import React, {useContext, useState} from 'react';
 
 import AppContext from '../../../AppContext.es';
+import {
+	EDIT_CUSTOM_OBJECT_FIELD,
+	dropLayoutBuilderField,
+} from '../../../actions.es';
 import Sidebar from '../../../components/sidebar/Sidebar.es';
+import DataLayoutBuilderContext from '../../../data-layout-builder/DataLayoutBuilderContext.es';
 import FieldsSidebarBody from './FieldsSidebarBody.es';
 import FieldsSidebarSettingsBody from './FieldsSidebarSettingsBody.es';
 import FieldsSidebarSettingsHeader from './FieldsSidebarSettingsHeader.es';
 
-export default function ({title}) {
-	const [{focusedCustomObjectField, focusedField}] = useContext(AppContext);
-	const [keywords, setKeywords] = useState('');
+export const DataEngineFieldsSidebar = ({title}) => {
+	const [dataLayoutBuilder] = useContext(DataLayoutBuilderContext);
+	const [
+		{
+			config,
+			customFields,
+			dataLayout,
+			editingLanguageId,
+			focusedCustomObjectField,
+			focusedField,
+		},
+		dispatch,
+	] = useContext(AppContext);
+
+	const hasFocusedCustomObjectField = (focusedCustomObjectField) => {
+		return !!focusedCustomObjectField.settingsContext;
+	};
 
 	const hasFocusedField = Object.keys(focusedField).length > 0;
-	const hasFocusedCustomObjectField =
-		Object.keys(focusedCustomObjectField).length > 0;
 
-	const displaySettings = hasFocusedCustomObjectField || hasFocusedField;
+	const displaySettings =
+		hasFocusedCustomObjectField(focusedCustomObjectField) ||
+		hasFocusedField;
+
+	const fieldTypes = dataLayoutBuilder
+		.getFieldTypes()
+		.filter(({group}) => group === 'basic');
+
+	fieldTypes.sort(({displayOrder: a}, {displayOrder: b}) => a - b);
```

Comment:
Could We avoid creating anonymous functions and create external and pure(defined outside the component)?